oneview/resource_server_certificate.go

```diff
@@ -0,0 +1,533 @@
+			BasicConstraints:   certificateDetails["basic_constraints"].(string),
+			Category:           certificateDetails["category"].(string),
+			CommonName:         certificateDetails["common_name"].(string),
+			ContactPerson:      certificateDetails["contact_person"].(string),
+			Country:            certificateDetails["country"].(string),
+			Created:            certificateDetails["created"].(string),
+			Description:        certificateDetails["description"].(string),
+			Dnqualifier:        certificateDetails["dn_qualifier"].(string),
+			Etag:               certificateDetails["etag"].(string),
+			Email:              certificateDetails["email"].(string),
+			EnhancedKeyUsage:   certificateDetails["enhanced_key_usage"].(string),
+			ExpiresInDays:      certificateDetails["expires_in_days"].(string),
+			GivenName:          certificateDetails["given_name"].(string),
+			Initials:           certificateDetails["initials"].(string),
+			Issuer:             certificateDetails["issuer"].(string),
+			KeyUsage:           certificateDetails["key_usage"].(string),
+			Locality:           certificateDetails["locality"].(string),
+			LocationState:      certificateDetails["location_state"].(string),
+			Modified:           certificateDetails["modified"].(string),
+			Name:               certificateDetails["name"].(string),
+			Organization:       certificateDetails["organization"].(string),
+			OrganizationalUnit: certificateDetails["organizational_unit"].(string),
+			PublicKey:          utils.NewNstring(certificateDetails["public_key"].(string)),
+			SerialNumber:       utils.NewNstring(certificateDetails["serial_number"].(string)),
+			Sha1Fingerprint:    utils.NewNstring(certificateDetails["sha1_fingerprint"].(string)),
+			Sha256Fingerprint:  utils.NewNstring(certificateDetails["sha256_fingerprint"].(string)),
+			Sha384Fingerprint:  utils.NewNstring(certificateDetails["sha384_fingerprint"].(string)),
+			SignatureAlgorithm: certificateDetails["signature_algorithm"].(string),
+			State:              certificateDetails["state"].(string),
+			Status:             certificateDetails["status"].(string),
+			Surname:            certificateDetails["surname"].(string),
+			Type:               certificateDetails["type"].(string),
+		}
+		CertificateDetailCollect = append(CertificateDetailCollect, servCCertificateDetails)
+		servC.CertificateDetails = CertificateDetailCollect
+	}
+	CertificateStatusList := d.Get("certificate_status").(*schema.Set).List()
+	for _, raw := range CertificateStatusList {
+		certificatestat := raw.(map[string]interface{})
+		servcCertificateStat := ov.CertificateStat{
+			ChainStatus: certificatestat["chain_status"].(string),
+			SelfSigned:  certificatestat["self_signed"].(bool),
+			Trusted:     certificatestat["trusted"].(bool),
+		}
+		servC.CertificateStatus = &servcCertificateStat
+	}
+
+	servCError := config.ovClient.UpdateServerCertificate(servC)
+	d.SetId(d.Get("name").(string))
```

Comment:
In create, Id is set with alias_name and here it's done with name. As per REST API documentation, GET calls are based on alias_name or IP address but not with a name. Please replace this name with alias_name and test this resource again.